### PR TITLE
[GPU] LSTMSequence and LSTMCell optimization

### PR DESCRIPTION
**Pull request by michal-miotk**
### Details:
 - creating simple primitive for lstm_sequence to be faster than previous approach using many primitives

### Tickets:
 - 146601

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/primitives/lstm.hpp [hunk 1]
```diff
@@ -145,4 +145,160 @@ struct lstm_elt : public primitive_base<lstm_elt> {
     }
 };
 
+
+struct lstm_seq : public primitive_base<lstm_seq> {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/lstm.hpp:149]: I suggest making it more generic and support different rnn kinds within 1 primitive

File: src/plugins/intel_gpu/src/kernel_selector/kernels/lstm/lstm_seq_kernel_base.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 6});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 1});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 2});
+    auto cldnnJit = GetJitConstants(orgParams);
+    auto entryPoint = GetEntryPoint(kernelName, orgParams.layerID, params);
+    auto jit = CreateJit(kernelName, cldnnJit, entryPoint);
+
+    kernel.params.workGroups.global = {out.X().v, out.Batch().v, 1};
+    kernel.params.workGroups.local = {out.X().v, 1, 1};
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/lstm/lstm_seq_kernel_base.cpp:84]: I think this is not safe as X may be larger than allowed LWS

Comment (michal-miotk) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/lstm/lstm_seq_kernel_base.cpp]: updated

File: src/plugins/intel_gpu/include/intel_gpu/primitives/lstm.hpp (hunk 1 above)
Comment (michal-miotk) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/lstm.hpp]: updated

File: src/plugins/intel_gpu/src/graph/impls/ocl/lstm_seq.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+#include "lstm_seq_inst.h"
+#include "lstm/lstm_seq_kernel_selector.h"
+#include "lstm/lstm_seq_kernel_base.h"
+#include "openvino/op/lstm_sequence.hpp"
+
+namespace cldnn {
+namespace ocl {
+
+struct lstm_seq_impl : typed_primitive_impl_ocl<rnn_seq> {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/impls/ocl/lstm_seq.cpp:15]: the impl should also be generic as well as ocl kernel

File: src/plugins/intel_gpu/src/plugin/ops/rnn.cpp [hunk 4]
```diff
@@ -224,127 +225,50 @@ static void CreateLSTMCellOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v4
...
+    cldnn::primitive_id lstm_seq_id = layerName;
+    auto mutable_precision_firstsecond = op->get_output_element_type(1);
 
-    if (!isForward) std::reverse(output_ids_offsets.begin(), output_ids_offsets.end());
-    // concatenated hidden state (output 1)
-    cldnn::primitive_id concatStr = layerName + ":hiddenConcat";
-    p.add_primitive(*op, cldnn::concatenation(concatStr, output_ids_offsets, 1));
+    if (p.use_new_shape_infer()) {
+        int direction = op->get_direction() == ov::op::RecurrentSequenceDirection::REVERSE ? 1 : 0;
+        cldnn::rnn_seq prim(layerName, inputs[0], inputs[1], \
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/ops/rnn.cpp:242]: Can it be applied to standalone LSTMCell operation too?

File: src/plugins/intel_gpu/src/graph/impls/ocl/lstm_seq.cpp (hunk 3 above)
Comment (michal-miotk) [reply src/plugins/intel_gpu/src/graph/impls/ocl/lstm_seq.cpp]: done

File: src/plugins/intel_gpu/src/plugin/ops/rnn.cpp (hunk 4 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/plugin/ops/rnn.cpp]: Is it? I still see decomposition of cell to subgraph

**Pull request closed without merging**